packages/apps/transfer/src/pages/faucet/existing/index.tsx

```diff
@@ -0,0 +1,189 @@
+import { IPollResponse } from '@kadena/chainweb-node-client';
+import { ICommandBuilder, IPactCommand, PactCommand } from '@kadena/client';
+import {
+  Button,
+  Heading,
+  SystemIcons,
+  TextField,
+} from '@kadena/react-components';
+
+import FormStatusNotification from './notification';
+
+import MainLayout from '@/components/Common/Layout/MainLayout';
+import { Select } from '@/components/Global';
+import { StyledOption } from '@/components/Global/Select/styles';
+import {
+  StyledAccountForm,
+  StyledForm,
+  StyledFormButton,
+} from '@/pages/transfer/cross-chain-transfer-finisher/styles';
+import { fundExistingAccount } from '@/services/faucet';
+import useTranslation from 'next-translate/useTranslation';
+import React, {
+  FC,
+  FormEvent,
+  FormEventHandler,
+  useCallback,
+  useState,
+} from 'react';
+
+// TODO: This needs to be changed to 100, when the contract is redeployed
+const AMOUNT_OF_COINS_FUNDED: number = 20;
+
+// eslint-disable-next-line @kadena-dev/typedef-var
+export const CHAINS = [
+  '0',
+  '1',
+  '2',
+  '3',
+  '4',
+  '5',
+  '6',
+  '7',
+  '8',
+  '9',
+  '10',
+  '11',
+  '12',
+  '13',
+  '14',
+  '15',
+  '16',
+  '17',
+  '18',
+  '19',
+] as const;
+
+export type ChainTuple = typeof CHAINS;
+export type Chain = ChainTuple[number];
+
+export type RequestStatus = 'not started' | 'pending' | 'succeeded' | 'failed';
+
+const ExistingAccountFaucetPage: FC = () => {
+  const { t } = useTranslation('common');
+
+  const [accountName, setAccountName] = useState('');
+  const [chainID, setChainID] = useState<Chain>('0');
+
+  const [requestStatus, setRequestStatus] = useState<{
+    status: RequestStatus;
+    message?: string;
+  }>({ status: 'not started' });
+
+  const onPoll = async (
+    transaction: IPactCommand & ICommandBuilder<Record<string, unknown>>,
+    pollRequest: Promise<IPollResponse>,
+  ): Promise<void> => {
+    const request = await pollRequest;
+    const result = request[transaction.requestKey!]?.result;
+    const status = result?.status;
+    if (status === 'failure') {
```

Comment:
shouldn't request statuses be defined somewhere?

Reply:
Discussed on Slack, `status` is an union type coming from `packages/libs/chainweb-node-client/src/interfaces/PactAPI.ts`